src/Imath/ImathVec.h

```diff
@@ -42,13 +45,60 @@ enum IMATH_EXPORT_ENUM InfException
 /// 2-element vector
 ///
 
-template <class T> class IMATH_EXPORT_TEMPLATE_TYPE Vec2
+namespace detail
+{
+template <typename T> struct Vec2StorageAdapter
+{
+    struct generated_store
+    {
+        union
+        {
+            struct
+            {
+                T x, y;
+            };
+            T arr[2];
+        };
+    };
+    using type = generated_store;
+};
+template <> struct Vec2StorageAdapter<uint8_t>
+{
+    using type = imath_v2b_t;
+};
+template <> struct Vec2StorageAdapter<int16_t>
+{
+    using type = imath_v2s_t;
+};
```

Comment:
Can this all be replaced by an alias template? Does this work?

    template<typename T> Vec2StorageAdapater {}
    template <> using Vec2StorageAdapter<uint8_t> = imath_v2b_t;
    template <> using Vec2StorageAdapter<uint16_t> = imath_v2s_t;
    ...

Reply:
oh, probably, except I needed to generate the one struct when there wasn't a "base" type that was serviceable - for user provided types, imagine saying Vec2<Vec2<f>> to declare a 2x2 matrix... so I went with the old school approach, but you're right that the alias should work for the ones that do exist